tests/Temporalio.Tests/Worker/WorkflowWorkerTests.cs

```diff
@@ -7395,6 +7396,95 @@ await ExecuteWorkerAsync<WorkflowUsingPriorities>(
+        {
+            await Workflow.WaitConditionAsync(() => complete);
+            await Workflow.ExecuteActivityAsync(() => SomeActivity(), new()
+            {
+                StartToCloseTimeout = TimeSpan.FromSeconds(10),
+            });
+        }
+
+        [WorkflowSignal]
+        public async Task CompleteAsync() => complete = true;
+
+        [Activity]
+        public static string SomeActivity() => "activity-result";
+    }
+
+    [Fact]
+    public async Task AutoscallingPollingWorker()
+    {
+        var promAddr = $"127.0.0.1:{TestUtils.FreePort()}";
+        var runtime = new TemporalRuntime(new()
+        {
+            Telemetry = new()
+            {
+                Metrics = new() { Prometheus = new(promAddr), },
+            },
+        });
+        var client = await TemporalClient.ConnectAsync(
+            new()
+            {
+                TargetHost = Client.Connection.Options.TargetHost,
+                Namespace = Client.Options.Namespace,
+                Runtime = runtime,
+            });
+
+        await ExecuteWorkerAsync<WaitOnSignalWorkflowThenActivity>(
+        async worker =>
+        {
+            await client.WorkflowService.GetSystemInfoAsync(new());
+
+            // Give pollers a beat to start
+            await Task.Delay(TimeSpan.FromMilliseconds(300));
+
+            using var httpClient = new HttpClient();
+            var resp = await httpClient.GetAsync(new Uri($"http://{promAddr}/metrics"));
+            var body = await resp.Content.ReadAsStringAsync();
+            var bodyLines = body.Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
+            var matches = bodyLines.Where(l => l.Contains("temporal_num_pollers")).ToList();
+            var activityPollers = matches.Where(l => l.Contains("activity_task")).ToList();
+
+            Assert.Single(activityPollers);
+            Assert.True(activityPollers[0].EndsWith('2'), "Activity poller count should be 2");
+            var workflowPollers = matches.Where(l => l.Contains("workflow_task")).ToList();
+            // Should have exactly two workflow poller metrics (sticky and non-sticky)
+            Assert.Equal(2, workflowPollers.Count);
+
+            // There's sticky & non-sticky pollers, and they may have a count of 1 or 2 depending on initialization timing
+            Assert.True(
+                workflowPollers[0].EndsWith('2') || workflowPollers[0].EndsWith('1'),
+                "First workflow poller count should be 1 or 2");
+            Assert.True(
+                workflowPollers[1].EndsWith('2') || workflowPollers[1].EndsWith('1'),
+                "Second workflow poller count should be 1 or 2");
+
+            // Run multiple workflow instances concurrently
+            var workflowTasks = Enumerable.Range(0, 20).Select(async _ =>
+            {
+                var handle = await client.StartWorkflowAsync(
+                    (WaitOnSignalWorkflow wf) => wf.RunAsync(),
```

Comment:
This seems to be starting a workflow that is not on this worker. After the signal below, you should wait on the workflow result if you expect it to finish